python/python/bystro/beanstalkd/worker.py

```diff
@@ -41,15 +42,46 @@
 logger = logging.getLogger(__name__)
 
 
+class FunctionWrapper:
+    def __init__(self, fn, queue):
+        self.fn_ser = cloudpickle.dumps(fn)
+        self.queue = queue
+
+    def __call__(self, *args, **kwargs):
+        worker_pid = os.getpid()
+        self.queue.put(worker_pid)
+
+        fn = cloudpickle.loads(self.fn_ser)
+        return fn(*args, **kwargs)
+
+
 # Signal handler function
+def kill_child_processes(parent_pid=os.getpid(), kill_parent=False):
```

Comment:
We need this to ensure that the process that`handler_fn` is running in, as well as any child processes handler_fn spawns are cleaned up either on program exit, or if touching the job fails with NOT_FOUND (or another exception), because if that happens the present worker no longer owns the job, and therefore the job is free to be picked up by another worker (applies when multiple workers are found; but even with a single worker we can get behavior that appears like multiple workers, because event message lifetimes are decoupled from submission message lifetimes).